subprojects/tooling-api/src/main/java/org/gradle/tooling/TestLauncher.java

```diff
@@ -27,6 +27,19 @@
  */
 public interface TestLauncher extends ConfigurableLauncher<TestLauncher> {
 
+    /**
+     * Specifies the test tasks to be executed.
+     * - When specified by default all tests are executed in the tasks
+     * - TODO how TestOperations overwrites this
+     * - TODO how withJVMTesttasks and methods overrides this
+     *
+     * @param testTasks
+     * @return this
+     * @since 6.1
+     */
+    @Incubating
+    TestLauncher withTests(String... testTasks);
```

Comment:
I don't see a `withJVMTesttasks`?

WDYT about making the semantics something like this:
- `withTests(TestOperationDescriptor...)` adds tests to execute.  We can find the right task from the test operation.
- `withJvmTestClasses(String...)` and `withJvmTestMethods(String, String...)` add tests to execute for _all_ `Test` tasks.
- New new methods like `withTaskAndClasses(Task testTask, String... testClasses)` and `withTaskAndMethods(Task testTask, String testClass, String... methods)` (think of better names) _add_ tests for the given task.

So if you do:
```
testLauncher.withTests(testOperationFromIntegTest)
testLauncher.withTaskAndClasses(crossVersionTestTask, "org.gradle.FooTest")
testLauncher.withJvmTestClasses("org.gradle.BarTest")
```

This would rerun the test from integTest, it would run `org.gradle.BarTest` from all `Test` tasks, and it would run `org.gradle.FooTest` from crossVersionTest.

Reply:
So instead of altering the behavior of the existing methods, we should introduce new ones to select the target test tasks. I think this is a better approach. 

In `withTaskAndClasses(Task testTask, String...)` what is the `Task` type? The `org.gradle.api.Task` cannot be used in the context of the Tooling API. Shall we use `org.gradle.tooling.model.Task`, or a `String` instead (i.e. the task path). The latter would be more convenient for JB as they already have a custom mapping of which test belongs to what tasks.

Reply:
Implemented with `String`s.